app/grandchallenge/reader_studies/models.py

```diff
@@ -1363,6 +1364,31 @@ def validate(  # noqa: C901
                 "Answer for required question cannot be None"
             )
 
+    @property
+    def html_string(self):
+        if self.question.answer_type in (
+            Question.AnswerType.CHOICE,
+            Question.AnswerType.MULTIPLE_CHOICE,
+            Question.AnswerType.MULTIPLE_CHOICE_DROPDOWN,
+        ):
+            list = [
+                format_html("<u>{}</u>", option)
+                if option in self.answer_text
+                else format_html("{}", option)
+                for option in self.question.options.values_list(
+                    "title", flat=True
+                ).all()
+            ]
+            return format_html(", ".join(list))
```

Comment:
This has the same effect - the first argument to `format_html` is marked safe implicitly, and you're including user input there. Can't this whole function be done in the template itself?

Reply:
Ok, but `format_html` also passes the elements through a conditional escape, so I thought that would be fine? I can move this to the template itself, of course. It just felt cleaner / easier to keep track of (especially as we potentially add more for annotations later) to do it in python.

Reply:
The first argument to `format_html` is not escaped, the rest are. Never, ever, send user input to the first argument of `format_html`, that should only be a raw string that we are in control of and are 100% sure is safe. https://docs.djangoproject.com/en/4.0/ref/utils/#django.utils.html.format_html

Just noticed that in the docs there is `format_html_join`, which could help you here https://docs.djangoproject.com/en/4.0/ref/utils/#django.utils.html.format_html_join

Reply:
Oh sorry, I see the difference now! `format_html_join` looks like a good solution. Thanks!

Reply:
`format_html_join` is not an option, it only works if all elements are formatted in the same way, which is not the case here (some are underlined, some are not). I will do it in the template. 

But just for my understanding: the elements in `list` are all individually constructed with `format_html`, so the user input, i.e. `option` in my code is escaped before it's marked safe (option is an argument to format_html). So then I'm not sure I understand why marking the joined string as safe is an issue.  Sorry for going on about this, but I want to make sure I fully understand this.

Reply:
Maybe in this specific case it's OK, but that would need to be covered by a lot of tests as you're relying on future developers not to touch `list` (btw, you should never use `list` as a variable name as it shadows the built-in `list` function). See the substitutions tests.

Honestly, what you're trying to do is too complex for a small snippet, this is much better solved in a template where all of these cases are handled for you.

Reply:
Ok, thanks for explaining. I'll move this to the template.